packages/cozy-client/src/store/mango.js

```diff
@@ -1,33 +1,6 @@
-const and = conditions => obj => {
-  for (let c of conditions) {
-    if (!c(obj)) {
-      return false
-    }
-  }
-  return true
-}
+import sift from 'sift'
 
-export const selectorFilter = selector => {
-  const filters = Object.keys(selector).map(attr => doc => {
-    const attrSel = selector[attr]
-    const val = doc[attr]
-    if (!(typeof attrSel === 'object')) {
-      const res = doc[attr] === attrSel
-      return res
-    }
-    const attrConditions = Object.keys(attrSel).map($operator => doc => {
-      const selValue = attrSel[$operator]
-      if ($operator == '$gt') {
-        return val > selValue
-      } else if ($operator == '$lt') {
-        return val < selValue
-      } else if ($operator == '$ne') {
-        return val !== selValue
-      } else {
-        throw new Error(`Unknown operator ${$operator}`)
-      }
-    })
-    return and(attrConditions)(doc)
-  })
-  return and(filters)
+export const selectorFilter = selector => doc => {
+  const test = sift(selector, [doc])
```

Comment:
It would be good if we did not have to wrap in an array.

Reply:
Looking at https://github.com/crcn/sift.js/tree/master#api, I think we could just have

```
export const selectorFilter = selector => sift(selector)
```

> array - sifts against target array. Without this, a function is returned

```js
var siftExists = sift({$exists:true});

siftExists('craig'); //true
siftExists(null); //false
['craig',null].filter(siftExists); //['craig']
```